metaflow/plugins/argo/argo_workflows.py

```diff
@@ -954,6 +1041,28 @@ def _container_templates(self):
                 )
             )
 
+            if node.name == "end":
+                if self._ignore_events:
+                    return
+                elif are_events_configured():
+                    template_env = {"METAFLOW_EVENT_SOURCE_NAME": EVENT_SOURCE_NAME}
+                    if EVENT_SOURCE_URL.startswith("nats://"):
+                        parsed = parse.urlparse(EVENT_SOURCE_URL)
+                        template_env["METAFLOW_EVENT_ENDPOINT"] = parsed.netloc
+                        template_env["METAFLOW_EVENT_PATH"] = parsed.path.replace(
+                            "/", ""
+                        )
+                    else:
+                        template_env["METAFLOW_EVENT_SOURCE_URL"] = EVENT_SOURCE_URL
+                    yield self._make_lifecycle_hook_container_template(
+                        template_env, "succeeded"
+                    )
+
+    def _make_lifecycle_hook_container_template(self, env, status):
+        t = WorkflowLifecycleHookContainerTemplate(current_flow_name(), status)
```

Comment:
All invocations of `current_flow_name` can be replaced with `self.name`

Reply:
Replaced `current_flow_name()` refs with `self.name` but left the separate function in place. I think it makes it easier to grok the overall flow.